src/OrleansLoggerProvider/LegacyOrleansLogger.cs

```diff
@@ -0,0 +1,149 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions.Internal;
+using Orleans.Runtime;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+
+namespace Orleans.Extensions.Logging.Legacy
+{
+    /// <summary>
+    /// LegacyOrleansLogger supports legacy orleans logging features, including <see cref="ILogConsumer"/>, <see cref="ICloseableLogConsumer">,
+    /// <see cref="IFlushableLogConsumer">, <see cref="Severity">. 
+    /// </summary>
+    [Obsolete("Use Microsoft.Extensions.Logging built-in loggers")]
+    public class LegacyOrleansLogger : ILogger
+    {
+        private readonly TimeSpan flushInterval = Debugger.IsAttached ? TimeSpan.FromMilliseconds(10) : TimeSpan.FromSeconds(1);
+        private DateTime lastFlush = DateTime.UtcNow;
+
+        private readonly IList<ILogConsumer> logConsumers;
+        private readonly string name;
+        private readonly IList<IFlushableLogConsumer> flushableConsumers;
+        private readonly LoggerType loggerType;
+        private readonly IPEndPoint ipEndPoint;
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="categoryName"></param>
+        /// <param name="logConsumers"></param>
+        /// <param name="maxSeverityLevel"></param>
+        public LegacyOrleansLogger(string categoryName, IList<ILogConsumer> logConsumers, IPEndPoint ipEndPoint)
+        {
+            this.logConsumers = logConsumers;
```

Comment:
We only need an enumerable, but not modifiable version, I'd choose another type instead of ```IList```, same for ```flushableConsumers```.

Specially as we are calling it like this: ```this.LogConsumers.ToArray()```